### PR TITLE
[10.0][FIX] sale_order_margin_percent: computation multi record exception

### PR DESCRIPTION
**Pull request by wtaferner**
Straight forward and easy to review and merge :-)

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

File: sale_order_margin_percent/models/sale_order_margin.py [hunk 1]
```diff
@@ -12,5 +12,6 @@ class SaleOrder(models.Model):
 
     @api.depends('margin', 'amount_untaxed')
     def _compute_percent(self):
-        if self.margin and self.amount_untaxed:
-            self.percent = (self.margin / self.amount_untaxed) * 100
+        for order in self:
+            if order.margin and order.amount_untaxed:
```
Comment (espo-tony) [inline sale_order_margin_percent/models/sale_order_margin.py]: IMO, this condition can be rewritten without the first term, as a margin equal to 0 would correctly compute a percent equal to 0.

File: sale_order_margin_percent/models/sale_order_margin.py [hunk 2]
```diff
@@ -12,5 +12,6 @@ class SaleOrder(models.Model):
...
+                order.percent = (order.margin / order.amount_untaxed) * 100
```
Comment (espo-tony) [inline sale_order_margin_percent/models/sale_order_margin.py]: I would add an else branch to handle the case with amount_untaxed equal to 0: in this case I think that percent should also be set to 0 (currently, if amount_untaxed was any number and lately drops to 0, the percent would simply not change, thus giving a wrong information).

**Review by espo-tony — CHANGES_REQUESTED** [review]

File: sale_order_margin_percent/models/sale_order_margin.py (hunk 1 above)
Comment (wtaferner) [reply sale_order_margin_percent/models/sale_order_margin.py]: It is not really part of this PR which is an actual bugfix waiting to be merged for quite some time although there is always space to improve. Not sure how it works here.

File: sale_order_margin_percent/models/sale_order_margin.py (hunk 2 above)
Comment (wtaferner) [reply sale_order_margin_percent/models/sale_order_margin.py]: Same as above.

**Comment by wtaferner** [comment]
@tafaRU 
Your review is welcome...as said, it is straight forward and waiting to be fixed and merged ;-)

**Comment by pedrobaeza** [comment]
Indeed it's a bugfix but it's very easy to include the else part. Any way, if not, you should at least change module version number.

**Comment by wtaferner** [comment]
Ok, I rewrote it that it will always provide the correct percentage and increased the version number. I hope this is what is expected and serves the input. Thank you for your reviews.

File: sale_order_margin_percent/models/sale_order_margin.py [hunk 3]
```diff
@@ -12,5 +12,8 @@ class SaleOrder(models.Model):
 
     @api.depends('margin', 'amount_untaxed')
```
Comment (wpichler) [inline sale_order_margin_percent/models/sale_order_margin.py:13]: adding @api.multi would be no failure.

**Comment by wpichler** [comment]
why not change the percent field to store=True ? I would make it searchable. Eg. Search for all sale orders with margin > 0.1

File: sale_order_margin_percent/models/sale_order_margin.py (hunk 3 above)
Comment (pedrobaeza) [reply sale_order_margin_percent/models/sale_order_margin.py:13]: On v10, @api.multi is the default

Comment (wpichler) [reply sale_order_margin_percent/models/sale_order_margin.py:13]: correct - but it does not hurt - and makes it clear on a short view

Comment (wtaferner) [reply sale_order_margin_percent/models/sale_order_margin.py:13]: I tend to skip this as it is only needed for public methods to work with RPC in v10 and we should move the same direction as Odoo core and save lines of code for something which is default behavior. We do not add ```store=True``` to every normal field, do we?

**Review by wtaferner — COMMENTED** [review]
Hmm, I do like the idea to make it searchable by storing the value in the database. Any contradictions @pedrobaeza?

**Comment by misern2** [comment]
👍  to search by margin.

**Comment by wtaferner** [comment]
Later PR and fix was merged (https://github.com/OCA/sale-workflow/pull/596). Therefore I close this PR. Thank you for your contributions and comments.

**Pull request closed without merging**